### PR TITLE
Initial Commit for Nyxus Recipe

### PR DESCRIPTION
**Pull request by sameeul**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

@conda-forge-admin, please ping conda-forge/help-c-cpp
@conda-forge//help-c-cpp, ready for review!

### CODE REVIEW

File: recipes/nyxus/meta.yaml
Comment (bollwyvl) [inline recipes/nyxus/meta.yaml:9]: one of the things from the deleted checklist in the issue template is about using official sources... you should try to do so if possible.

File: recipes/nyxus/meta.yaml
Comment (bollwyvl) [inline recipes/nyxus/meta.yaml:22]: anything python-related should _probably_ be in `host` rather than build

File: recipes/nyxus/meta.yaml
Comment (bollwyvl) [inline recipes/nyxus/meta.yaml:37]: should just be `python`, `conda-build` takes care of matrixing-out version stuff

File: recipes/nyxus/meta.yaml
Comment (bollwyvl) [inline recipes/nyxus/meta.yaml:43]: if it actually builds against numpy, there are some tricks:

https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

File: recipes/nyxus/meta.yaml
Comment (bollwyvl) [inline recipes/nyxus/meta.yaml:32]: `numpy` and `pandas` have pins: it's good to respect them:

https://github.com/PolusAI/nyxus/blob/0.2.3/setup.py#L110

File: recipes/nyxus/bld.bat
Comment (bollwyvl) [inline recipes/nyxus/bld.bat:1]: if it's possible to build without GPU support, it can often be worth it... it can be a bit tricky, though, to handle the metapackages properly...

File: recipes/nyxus/build.sh
Comment (bollwyvl) [inline recipes/nyxus/build.sh:7]: some of these are available from https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html#environment-variables-set-during-the-build-process

File: recipes/nyxus/LICENSE.txt
Comment (bollwyvl) [inline recipes/nyxus/LICENSE.txt:1]: this should come from the upstream tarball, if at all possible

File: recipes/nyxus/build.sh
Comment (bollwyvl) [inline recipes/nyxus/build.sh:9]: if possible, you should emulate the example invocation, e.g


```
python -m pip install .
```

this creates better metadata than calling `setup.py` directly

File: recipes/nyxus/build.sh
Comment (bollwyvl) [inline recipes/nyxus/build.sh:3]: `set -eux` is a bit stricter and will give you more feedback

File: recipes/nyxus/meta.yaml
Comment (sameeul) [reply recipes/nyxus/meta.yaml:9]: Yes, we will move to official source as soon as one of my PR gets merged.

File: recipes/nyxus/meta.yaml
Comment (sameeul) [reply recipes/nyxus/meta.yaml:22]: Got it!

File: recipes/nyxus/bld.bat
Comment (sameeul) [reply recipes/nyxus/bld.bat:1]: So, we want to build with CUDA support eventually. What is the best way to do that. I am thinking about adding cudatoolkit-dev as a build dependency and cudatoolkit as run dependency.

File: recipes/nyxus/LICENSE.txt
Comment (sameeul) [reply recipes/nyxus/LICENSE.txt:1]: The upstream actually has the license. I will remove it from here.

File: recipes/nyxus/meta.yaml
Comment (carterbox) [reply recipes/nyxus/meta.yaml:9]: Are your changes required for this recipe to build on conda-forge? Use a patch instead of publishing from an unofficial repository.

Comment (sameeul) [reply recipes/nyxus/meta.yaml:9]: My changes will get merged to the official repo today or tomorrow. As soon as that gets merged, I plan to update this. I resorted to this quick hack since I need to change both the build process in the actual repo and test it out with the recipe.

File: recipes/nyxus/meta.yaml
Comment (bollwyvl) [inline recipes/nyxus/meta.yaml:56]: probably add `pip` to `requires` and `pip check` to `commands`.

File: recipes/nyxus/meta.yaml
Comment (bollwyvl) [inline recipes/nyxus/meta.yaml:48]: could also do with a minimum `pandas` bottom pin matching upstream

File: recipes/nyxus/meta.yaml
Comment (sameeul) [reply recipes/nyxus/meta.yaml:56]: so, like the following?
```
  requires:
    - pytest
    - pip
  imports:
    - nyxus
  commands:
    - pip check
    - python -m pytest run_test.py
```

File: recipes/nyxus/bld.bat
Comment (carterbox) [reply recipes/nyxus/bld.bat]: Do not use cudatoolkit-dev in a recipe. Just add `compiler('cuda')` to the build requirements section. Add the cudatoolkit to your host requirements if you need the CUDA libraries.

File: recipes/nyxus/meta.yaml
Comment (carterbox) [inline recipes/nyxus/meta.yaml:49]:
```suggestion
```

File: recipes/nyxus/run_test.py
Comment (carterbox) [inline recipes/nyxus/run_test.py:1]: https://docs.conda.io/projects/conda-build/en/latest/concepts/recipe.html

This file is run twice because it is named run_test.[py] and also you call it in the test/commands section. Is it even needed? The import of nyxus is also tested in test/imports. Remove this file if it is redundant.

File: recipes/nyxus/meta.yaml
Comment (carterbox) [inline recipes/nyxus/meta.yaml:38]:
```suggestion
    - blosc  # [cuda_compiler_version == "None"]
```
It looks like blosc is unused in CUDA builds because the link inspector says there are no links to it.

File: recipes/nyxus/meta.yaml
Comment (carterbox) [inline recipes/nyxus/meta.yaml:19]:
```suggestion
build:
  number: 0
  skip: true  # [py<36]
  missing_dso_whitelist:
    - "*/nvcuda.dll"   # [win]
  ignore_run_exports_from:
  # The CXX standard library is unused on linux, but used on osx
    - {{ compiler('cxx') }}  # [linux]
```
According to the link inspector the cxx standard library is unused on linux.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/nyxus/meta.yaml
Comment (sameeul) [reply recipes/nyxus/meta.yaml:41]: This I believe only happens for CUDA 10.2. The build machine uses GCC 7.5 which fails the minimum GCC requirement for `xtensor`. There is a open issue in the `xtensor` Github about this. In this particular situation, we build Nyxus w/o zarr file support. CUDA 11.x uses blosc. So, should I use `# [cuda_compiler_version == "None" or cuda_compiler_version == "10.2.0"] ` ?

File: recipes/nyxus/run_test.py
Comment (sameeul) [reply recipes/nyxus/run_test.py]: Thanks for the explanation. Removed the file.

File: recipes/nyxus/meta.yaml
Comment (sameeul) [reply recipes/nyxus/meta.yaml]: @carterbox : I updated meta.yaml and CMakeLists.txt so that it does not used bring in blosc in CUDA 10.2 build (where it is not used). I think the link inspector now does not find it anymore.

**Comment by sameeul** [comment]
@conda-forge-admin, please ping conda-forge/help-c-cpp @conda-forge//help-python
@conda-forge//help-c-cpp, ready for review!
@conda-forge//help-python, ready for review!

File: recipes/nyxus/meta.yaml
Comment (carterbox) [reply recipes/nyxus/meta.yaml]: > So, should I use # [cuda_compiler_version != "10.2.0"] ?

Yes, something like that is fine. Could have also dropped builds before 11.

**Pull request merged**
